src/main/java/org/spongepowered/common/mixin/core/item/inventory/MixinContainer.java

```diff
@@ -135,6 +138,18 @@ public void onPutStackInSlot(int slotId, ItemStack itemstack, CallbackInfo ci) {
         }
     }
 
+    @Inject(method = "slotClick", at = @At(value = "HEAD"), cancellable = true)
+    public void onSlotClick(int slotId, int clickedButton, int mode, EntityPlayer player, CallbackInfoReturnable<net.minecraft.item.ItemStack> ci) {
+        if (slotId >= inventorySlots.size()) {
+            // Kick the player who's sending invalid packets and cancel the click
```

Comment:
Less than zero is used in vanilla to let -999 be dropping/destroying.
Vanilla handles the validity for <0 properly.

On Sun, Jan 31, 2016, 6:47 PM Mumfrey <notifications@github.com> wrote:

> In
> src/main/java/org/spongepowered/common/mixin/core/item/inventory/MixinContainer.java
> <https://github.com/SpongePowered/SpongeCommon/pull/466#discussion_r51371303>
> :
>
> > @@ -135,6 +138,18 @@ public void onPutStackInSlot(int slotId, ItemStack itemstack, CallbackInfo ci) {
> >          }
> >      }
> >
> > +    @Inject(method = "slotClick", at = @At(value = "HEAD"), cancellable = true)
> > +    public void onSlotClick(int slotId, int clickedButton, int mode, EntityPlayer player, CallbackInfoReturnable<net.minecraft.item.ItemStack> ci) {
> > +        if (slotId >= inventorySlots.size()) {
> > +            // Kick the player who's sending invalid packets and cancel the click
>
> again, slotId is also invalid if it's less than zero
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/SpongePowered/SpongeCommon/pull/466/files#r51371303>.
>